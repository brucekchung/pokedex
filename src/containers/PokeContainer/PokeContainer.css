.pokemon {
  color: blue;
  display: flex;
  flex-wrap: wrap;
  margin: 0 auto;
  justify-content: space-around;
}

.PokeContainer {
  text-align: center;
}