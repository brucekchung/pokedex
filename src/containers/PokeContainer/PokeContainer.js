import React, { Component } from 'react'
import { shape, func, string } from 'prop-types' //PropTypes
import { connect } from 'react-redux'
import { sendTypeAction } from '../../actions/actionIndex'
import { getPokeType } from '../../api'
import { Loading } from '../../components/Loading/Loading'

export class PokeContainer extends Component {
  componentDidMount = async () => {
    const pokeType = await getPokeType()
    console.log('pokeType: ', pokeType)

    this.props.sendType(pokeType)
  }

  handleClick = () => {

    console.log('handleClick')
  }

  render() {
    return (
      <div>
        <Loading />
        <button onClick={this.handleClick}> Pokemon! </button>
      </div>
    );
  }
}

PokeContainer.propTypes = {
  fake: shape({ fake: string }),
  fakeAction: func //isRequired
};

export const mapState = ({ fake }) => ({ fake })

export const mapDispatch = dispatch => ({ 
  sendType: (type) => dispatch(sendTypeAction(type))
})

export default connect(mapState, mapDispatch)(PokeContainer)
