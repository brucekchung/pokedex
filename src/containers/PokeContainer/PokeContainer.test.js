import React from 'react'
import { PokeContainer, mapState, mapDispatch } from './PokeContainer'
import { shallow } from 'enzyme'

describe('PokeContainer', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<PokeContainer />)

    expect(wrapper).toMatchSnapshot()
  })

  it.skip('mapState should return the correct data', () => {

  })

  it.skip('mapDispatch should call dispatch when one of the props are called', () => {
    
  })
})