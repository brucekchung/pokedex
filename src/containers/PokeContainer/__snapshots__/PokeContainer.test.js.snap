// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PokeContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown />,
        <button
          onClick={[Function]}
        >
           Pokemon! 
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Pokemon! ",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": " Pokemon! ",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown />,
          <button
            onClick={[Function]}
          >
             Pokemon! 
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Pokemon! ",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": " Pokemon! ",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
